### PR TITLE
Fixing some TODOs and Iterator refactor

### PR DESCRIPTION
**Pull request by zotho**

### CODE REVIEW

**Comment by sdroege** [comment]
Looks good to me otherwise, thanks a lot for the cleanup!

**Comment by sdroege** [comment]
I commented on the individual commits btw

**Comment by zotho** [comment]
> `unwrap_or(0)` seems more easy to understand here (and is even shorter)

Agree. Replaced with `unwrap_or(0)`.

**Comment by sdroege** [comment]
Can you get rid of the `Revert iterator refactor` / `Iterator refactor` commits by rebasing? They don't give anything useful in the commit history :)

**Comment by zotho** [comment]
@sdroege Got rid of `Iterator refactor`, `Revert iterator refactor`, `Use assert_eq instead assert`, `unwrap_or_default fix` commits. Should look cleaner.

**Review by sdroege — APPROVED** [review]

**Comment by sdroege** [comment]
Thanks!

File: src/filter.rs [hunk 1]
```diff
@@ -239,10 +229,7 @@ impl Filter {
                         + b[3] * filter_state[3]
                         + b[4] * filter_state[4];
 
-                    filter_state[4] = filter_state[3];
-                    filter_state[3] = filter_state[2];
-                    filter_state[2] = filter_state[1];
-                    filter_state[1] = filter_state[0];
+                    filter_state.copy_within(0..4, 1);
```
Comment (sdroege) [inline src/filter.rs:232]: Note that this is only compiled to the same code with `-C opt-level=2`. With `-C opt-level=1` it ends up not being inlined, which then keeps bounds checks etc.

Comment (sdroege) [reply src/filter.rs:232]: As it turns out, even if inlined it's not compiling to the same code and causes a 20-35% slowdown. See https://github.com/sdroege/ebur128/issues/52

Reverted this part for now :)

Comment (zotho) [reply src/filter.rs:232]: Ok, I see. Thanks

**Pull request merged**